cvmfs/catalog_mgr_rw_impl.h

```diff
@@ -0,0 +1,181 @@
+/**
+ * This file is part of the CernVM file system.
+ */
+
+
+#ifndef CVMFS_CATALOG_MGR_RW_IMPL_H_
+#define CVMFS_CATALOG_MGR_RW_IMPL_H_
+
+#include "catalog_mgr_rw.h"
+
+#include <inttypes.h>
+#include <unistd.h>
+
+#include <cassert>
+#include <cstdio>
+#include <cstdlib>
+#include <ctime>
+#include <string>
+#include <vector>
+
+#include "catalog_rw.h"
+#include "directory_entry.h"
+#include "hash.h"
+#include "logging.h"
+#include "manifest.h"
+#include "smalloc.h"
+#include "statistics.h"
+#include "upload.h"
+#include "util.h"
+
+
+
+using namespace std;  // NOLINT
+
+namespace catalog {
+
+template <class CatalogMgrT>
+DirectoryEntryBase
+CatalogBalancer<CatalogMgrT>::CreateEmptyContentDirectoryEntryBase(
+    string name, uid_t uid, gid_t gid) {
+  shash::Algorithms algorithm = catalog_mgr_->spooler_->GetHashAlgorithm();
+  shash::Any file_hash(algorithm);
+  shash::HashString("", &file_hash);
+
+  DirectoryEntryBase deb;
+  deb.name_ = NameString(name);
+  deb.mode_ = S_IFREG | S_IRWXU;
+  deb.size_ = 0;
+  deb.linkcount_ = 1;
+  deb.checksum_ = file_hash;
+  deb.has_xattrs_ = false;
+  deb.mtime_ = time(NULL);
+  deb.inode_ = catalog::DirectoryEntry::kInvalidInode;
+  deb.parent_inode_ = catalog::DirectoryEntry::kInvalidInode;
+  deb.uid_ = uid;
+  deb.gid_ = gid;
+  return deb;
+}
+
+template <class CatalogMgrT>
+void CatalogBalancer<CatalogMgrT>::AddCvmfsCatalogFile(string path) {
```

Comment:
We call these usually `CatalogMarker` to avoid confusion with the file that actually contains the SQlite catalog.